sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -1112,4 +1112,15 @@ class SparkSqlAstBuilder extends AstBuilder {
       withIdentClause(ctx.identifierReference(), UnresolvedNamespace(_)),
       cleanedProperties)
   }
+
+  /**
+   * Create a [[ShowCollationsCommand]] command.
+   * Expected format:
+   * {{{
+   *   SHOW COLLATIONS (LIKE? pattern)?;
+   * }}}
+   */
+  override def visitShowCollations(ctx: ShowCollationsContext): LogicalPlan = withOrigin(ctx) {
```

Comment:
Could you take a look at what `visitShowFunctions` does? I believe that approach might be the best for us, as in that case we have a clear distinction of what collation is being listed and we can easily remove addition of catalog and schema fields from CollationFactory.

Reply:
Also, could we move than this function to AstBuilder, where visitShowFunctions is as well

Reply:
Do we need to define the syntax of `SHOW COLLATIONS` as follows?
```
| SHOW identifier? COLLATIONS ((FROM | IN) ns=identifierReference)?
        (LIKE? pattern=stringLit)?                                     #showCollations
```

Reply:
We can do two things here. Follow completely functions implementation, but just throw a new error that user defined collations are not supported, or just keep SHOW COLLATIONS (LIKE? pattern=stringLit)? and make sure we implement the listing of collation in a similar way as functions. Second approach might be more clear for now, as we would just throw parse exception if they tried to use some user defined collation specific implementation. ((FROM | IN) ns=identifierReference)? part of the parsing is a good add-on, but for now we are not going to use it. In showFunctions we only use identifier? and ((FROM | IN) ns=identifierReference)? when we are listing udfs and temporary functions from specific catalogs/schemas.

Reply:
If you ask me I would keep SHOW COLLATIONS (LIKE? pattern=stringLit)? but try to implement the underlying stuff as for functions. The addition of other features will not be that hard if we follow the functions implementation now.

Reply:
Let me implement a version first (`Follow completely functions implementation`). 
From my perspective, it seems a bit reasonable. 
Let's take a look at its final appearance. 
If you think it's okay, we'll do it. If you don't think it's good, I'll update it later.

Reply:
> Also, could we move than this function to AstBuilder, where visitShowFunctions is as well

Because `ShowCollationsCommand` inherits `UnaryRunnableCommand`, and `UnaryRunnableCommand` is in module `sql/core`, So we need to put `visitShowCollations` in `SparkSqlParser` instead of `AstBuilder`,
https://github.com/apache/spark/pull/47038#discussion_r1651878633

Reply:
I agree with that. What do you think about using ShowCollations as a new LogicalPlan node and then resolve that to either ShowCollationsCommand or ShowCollationsExec, as ShowCollationsExec represents a v1 path of execution. (Something similar was done for Functions)

Reply:
I'm not sure if this is reasonable, but based on the logic that has already been implemented, is it sufficient? I would like to ask @MaxGekk , what do you think? I would like to hear your advice.
Thanks!